_posts/2023-10-03-securing-wildfly-apps-oidc-openshift.adoc

```diff
@@ -0,0 +1,360 @@
+
+== Prerequisites
+
+To follow along with this guide, you will need:
+
+* Roughly 15 minutes
+* Access to an OpenShift cluster (try the https://developers.redhat.com/developer-sandbox[Red Hat Developer Sandbox] for free)
+* https://docs.openshift.com/container-platform/4.13/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]
+* https://docs.wildfly.org/29/Getting_Started_on_OpenShift.html#helm-charts[WildFly Helm Chart]
+
+== Example Application
+
+We will use a simple web application in this guide that consists of a single https://github.com/wildfly-security/elytron-examples/blob/main/simple-webapp-oidc/src/main/java/org/wildfly/security/examples/SecuredServlet.java[servlet]. We will secure this servlet using OIDC.
+
+We will use the example in the https://github.com/wildfly-security-incubator/elytron-examples/tree/main/simple-webapp-oidc[simple-webapp-oidc] directory in this repo.
+
+To obtain this example, clone the elytron-examples repository to your local machine:
+
+[source]
+----
+git clone git@github.com:wildfly-security-incubator/elytron-examples.git
+----
+
+== Log Into the OpenShift Cluster
+
+Before we can deploy our application, we need to log in to an OpenShift cluster. You can log in via the https://docs.openshift.com/container-platform/4.13/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]:
+
+[source]
+----
+oc login -u myUserName
+----
+
+Alternatively, you can log in using an API token:
+
+[source]
+----
+oc login --token=myToken --server=myServerUrl
+----
+
+You can request the token via the *Copy Login Command* link in the OpenShift web console.
+
+If you don't already have a project created, you can create one using:
+
+[source]
+----
+oc new-project myProjectName
+----
+
+== Start Keycloak
+
+We will be using Keycloak as our OIDC identity provider.
+
+To start a Keycloak server in your project on OpenShift, use the following command:
+
+[source]
+----
+oc process -f https://raw.githubusercontent.com/keycloak/keycloak-quickstarts/latest/openshift/keycloak.yaml \
+    -p KEYCLOAK_ADMIN=admin                  \// <1>
+    -p KEYCLOAK_ADMIN_PASSWORD=admin         \// <2>
+    -p NAMESPACE=<PROJECT_NAME>                \// <3>
+| oc create -f -
+----
+<1> Replace `admin` with the user name you would like to use when accessing the Keycloak Administration Console.
+<2> Replace `admin` with the password you would like to use when accessing the Keycloak Administration Console.
+<3> Replace `<PROJECT_NAME>` with your project name.
+
+After running the above command, you should see the following output:
+
+[source]
+----
+service/keycloak created
+route.route.openshift.io/keycloak created
+Warning: apps.openshift.io/v1 DeploymentConfig is deprecated in v4.14+, unavailable in v4.10000+
```

Comment:
@PrarthonaPaul Would you be able to also check if you're seeing this warning too?

Reply:
Trying it out now!

Reply:
Prarthona Paul: It seems that for OpenShift 4.14 and later, they are deprecating deploymentConfig and are encouraging customers to use Deployment instead: https://access.redhat.com/articles/7041372
I see that we are using DeploymentConfig in [keycloak.yaml. ](https://raw.githubusercontent.com/keycloak/keycloak-quickstarts/latest/openshift/keycloak.yaml)But maybe we can use the Deployment object instead of that unless we need something specific that only DeploymentConfig offers? 
Developer sandbox uses OpenShift version 4.14.1. That's why the warning is there

Reply:
That's great info, thanks Prarthona! Would you be able to create a Keycloak github issue here (if one doesn't already exist for this)?

https://github.com/keycloak/keycloak/issues

Reply:
Done!

Reply:
Great, please add the link to the issue here so we have it for reference later on. Thanks!

Reply:
here you go: https://github.com/keycloak/keycloak/issues/24607